File: dev/20190402/k2works/sam-app/HelloWorldFunction/src/test/java/fizzbuzz/FizzBuzzTest.java

package fizzbuzz;

import org.junit.jupiter.api.BeforeEach;
import org.junit.jupiter.api.Test;

import static org.junit.jupiter.api.Assertions.assertEquals;

public class FizzBuzzTest {
    private FizzBuzzValueCommand _fizzBuzzValueCommand;
    private FizzBuzzValuesCommand _fizzBuzzValuesCommand;

    @BeforeEach
    void init() {
        _fizzBuzzValueCommand = new FizzBuzzValueCommand(FizzBuzzType.valueOf("one"));
        _fizzBuzzValuesCommand = new FizzBuzzValuesCommand(FizzBuzzType.valueOf("one"));
    }

    @Test
    public void 値を1から100までをプリントする() {
        _fizzBuzzValuesCommand.execute(100);
        assertEquals("1", _fizzBuzzValuesCommand.getValues().get(0).getValue());
        assertEquals("Fizz", _fizzBuzzValuesCommand.getValues().get(2).getValue());
        assertEquals("Buzz", _fizzBuzzValuesCommand.getValues().get(4).getValue());
        assertEquals("FizzBuzz", _fizzBuzzValuesCommand.getValues().get(14).getValue());
        assertEquals("Buzz", _fizzBuzzValuesCommand.getValues().get(99).getValue());
    }

    @Test
    public void 値が3で割り切れる場合はFizzをプリントする() {
        _fizzBuzzValueCommand.execute(3);
        assertEquals("Fizz", _fizzBuzzValueCommand.getValue());
        _fizzBuzzValueCommand.execute(1);
        assertEquals("1", _fizzBuzzValueCommand.getValue());
    }

    @Test
    public void 値が5で割り切れる場合はBuzzをプリントする() {
        _fizzBuzzValueCommand.execute(5);
        assertEquals("Buzz", _fizzBuzzValueCommand.getValue());
    }

    @Test
    public void 値が3と5で割り切れる場合はFizzBuzzをプリントする() {
        _fizzBuzzValueCommand.execute(15);
        assertEquals("FizzBuzz", _fizzBuzzValueCommand.getValue());
    }

    @Test
    public void タイプ1は通常のパターンを返す() {
        FizzBuzzValueCommand fizzBuzzValueCommand = new FizzBuzzValueCommand(FizzBuzzType.valueOf("one"));
        FizzBuzzValuesCommand fizzBuzzValuesCommand = new FizzBuzzValuesCommand(FizzBuzzType.valueOf("one"));
        fizzBuzzValueCommand.execute(3);
        assertEquals("Fizz", fizzBuzzValueCommand.getValue());
        fizzBuzzValuesCommand.execute(100);
        assertEquals("Buzz", fizzBuzzValuesCommand.getValues().get(4).getValue());
        assertEquals("FizzBuzz", fizzBuzzValuesCommand.getValues().get(14).getValue());
    }

    @Test
    public void タイプ2は数字のみのパターンを返す() {
        FizzBuzzValueCommand fizzBuzzValueCommand = new FizzBuzzValueCommand(FizzBuzzType.valueOf("two"));
        FizzBuzzValuesCommand fizzBuzzValuesCommand = new FizzBuzzValuesCommand(FizzBuzzType.valueOf("two"));
        fizzBuzzValueCommand.execute(3);
        assertEquals("3", fizzBuzzValueCommand.getValue());
        fizzBuzzValuesCommand.execute(100);
        assertEquals("5", fizzBuzzValuesCommand.getValues().get(4).getValue());
        assertEquals("15", fizzBuzzValuesCommand.getValues().get(14).getValue());
    }

    @Test
    public void タイプ3はFizzBuzzのみのパターンを返す() {
        FizzBuzzValueCommand fizzBuzzValueCommand = new FizzBuzzValueCommand(FizzBuzzType.valueOf("three"));
        FizzBuzzValuesCommand fizzBuzzValuesCommand = new FizzBuzzValuesCommand(FizzBuzzType.valueOf("three"));
        fizzBuzzValueCommand.execute(3);
        assertEquals("3", fizzBuzzValueCommand.getValue());
        fizzBuzzValuesCommand.execute(100);
        assertEquals("5", fizzBuzzValuesCommand.getValues().get(4).getValue());
        assertEquals("FizzBuzz", fizzBuzzValuesCommand.getValues().get(14).getValue());
    }
}

File: dev/20190510/session/src/FizzBuzzTest.php
<?php

use PHPUnit\Framework\TestCase;
class FizzBuzzTest extends TestCase {
  /** @noinspection NonAsciiCharacters */
  /**
  * @test
  */
  public function 値を1から100までプリントする() {
      $list = printValue();
      $this->assertEquals(1, $list[0]);
      $this->assertEquals(100, $list[99]);
  }
  /** @noinspection NonAsciiCharacters */
  /**
  * @test
  */
  public function 値が3で割り切れる場合はFizzをプリントする() {
      $this->assertEquals("Fizz", fizzBuzz(3));
  }
  /** @noinspection NonAsciiCharacters */
  /**
  * @test
  */
  public function 値が5で割り切れる場合はBuzzをプリントする() {
      $this->assertEquals("Buzz", fizzBuzz(5));
  }
  /** @noinspection NonAsciiCharacters */
  /**
  * @test
  */
  public function 値が3と5の両方で割り切れる場合はFizzBuzzをプリントする(){
      $this->assertEquals("FizzBuzz",fizzBuzz(15));
  }
}

function fizzBuzz($number) {
    $result = $number;
    if ($number % 3 === 0) {
        $result = "Fizz";
    }
    if ($number % 5 === 0){
        $result = "Buzz";
    }
    if($number % 3 === 0 && $number % 5 === 0){
        $result = "FizzBuzz";
    }
    return $result;
}

function printValue() {
    $list = [];
    for($i = 1; $i <= 100; $i++) {
       $list[] = $i; 
    }
    return $list;
}